core/src/main/java/org/geysermc/geyser/util/EntityUtils.java

```diff
@@ -209,6 +209,18 @@ public static void updateMountOffset(Entity passenger, Entity mount, boolean rid
                         zOffset = displayTranslation.getZ();
                     }
                 }
+                case PLAYER -> {
```

Comment:
I'd bet that the same is occurring for text displays with multiple lines mounted on anything, not just players - I wonder if this is even the proper spot to add this

Reply:
You mean moving the entire if outside the switch?

Reply:
For the purpose of this PR, this is probably fine; but should be fixed properly sometime in the future